<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="BreakSeq2">
    <meta name="author" content="Marghoob Mohiyuddin">

    <title>BreakSeq2</title>

    <!-- Bootstrap core CSS -->
    <link href="http://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
    <style>
    .container {
  margin-right: auto;
  margin-left: auto;
  max-width: 760px;
}
    </style>
   
    
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
       m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
         })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-58409498-4', 'auto');
  ga('require', 'linkid', 'linkid.js');
    ga('send', 'pageview');


  
/**
* Function that tracks a click on an outbound link in Google Analytics.
* This function takes a valid URL string as an argument, and uses that URL string
* as the event label.
*/
var trackOutboundLink = function(url) {
   ga('send', 'event', 'outbound', 'click', url, {'hitCallback':
     function () {
     document.location = url;
     }
   });
}


function _gaLt(event){
    var el = event.srcElement || event.target;

    /* Loop up the DOM tree through parent elements if clicked element is not a link (eg: an image inside a link) */
    while(el && (typeof el.tagName == 'undefined' || el.tagName.toLowerCase() != 'a' || !el.href)){
        el = el.parentNode;
    }

    if(el && el.href){
        if(el.href.indexOf(location.host) == -1){ /* external link */
            /* HitCallback function to either open link in either same or new window */
            var hitBack = function(link, target){
                target ? window.open(link, target) : window.location.href = link;
            };
            /* link */
            var link = el.href;
            /* Is target set and not _(self|parent|top)? */
            var target = (el.target && !el.target.match(/^_(self|parent|top)$/i)) ? el.target : false;
            /* send event with callback */
            ga(
                "send", "event", "Outgoing Links", link,
                document.location.pathname + document.location.search,
                {"hitCallback": hitBack(link, target)}
            );

            /* Prevent standard click */
            event.preventDefault ? event.preventDefault() : event.returnValue = !1;
        }

    }
}

/* Attach the event to all clicks in the document after page has loaded */
var w = window;
w.addEventListener ? w.addEventListener("load",function(){document.body.addEventListener("click",_gaLt,!1)},!1)
 : w.attachEvent && w.attachEvent("onload",function(){document.body.attachEvent("onclick",_gaLt)});

</script>

</head>

<body>

	

    <div class="container" style="padding-bottom: 400px;">
    	<div class="page-header">
  		<h1>BreakSeq2</h1>
  		<h2>Ultrafast and accurate nucleotide-resolution analysis of structural variants</h2>
	</div>
    
        <div class="jumbotron">
            <p>
                <i>For more information contact us at <a href="mailto:rd@bina.com">rd@bina.com</a></i>
            </p>
            
        </div>
        
    <h2>Publication</h2>

    <div class="panel panel-default" style="font-family:monospace;">
        <div class="panel-body">
            <i>If you use BreakSeq2 in your work, please cite the following:</i><br>
            Alexej Abyzov, Shantao Li, Daniel Rhee Kim, Marghoob Mohiyuddin, Adrian M. St√ºtz, Nicholas F. Parrish, Xinmeng Jasmine Mu, Wyatt Clark, Ken Chen, Matthew Hurles, Jan O. Korbel, Hugo Y. K. Lam, Charles Lee and Mark B. Gerstein<br>
            <b>Analysis of deletion breakpoints from 1,092 humans reveals details of mutation mechanisms</b><br>
            Nature CommunicationsBioinformatics published June 1, 2015 <a
                href="http://www.nature.com/ncomms/2015/150601/ncomms8256/full/ncomms8256.html"
                onclick="trackOutboundLink('http://www.nature.com/ncomms/2015/150601/ncomms8256/full/ncomms8256.html'); return false;">http://www.nature.com/ncomms/2015/150601/ncomms8256/full/ncomms8256.html</a>
        </div>
    </div>

<h2>Download BreakSeq2</h2>

<p>Latest version:  <a href="https://github.com/bioinform/breakseq2/archive/2.1-alpha.tar.gz"
onclick="trackOutboundLink('https://github.com/bioinform/breakseq2/archive/2.1-alpha.tar.gz'); return false;">https://github.com/bioinform/breakseq2/archive/2.1-alpha.tar.gz</a></p>

<p>For other versions, see "releases". <a href="https://github.com/bioinform/breakseq2/releases"
onclick="trackOutboundLink('https://github.com/bioinform/breakseq2/releases'); return false;">https://github.com/bioinform/breakseq2/releases</a></p>


<h2>System Requirements</h2>
<p>
   The following Python packages must be installed:
   <ul>
      <li><a href="https://pypi.python.org/packages/source/p/pysam/pysam-0.7.7.tar.gz">pysam-0.7.7</a>: BreakSeq2 has only been tested with version 0.7.7 of pysam.</li>
      <li><a href="http://biopython.org/wiki/Main_Page">Biopython</a></li>
   </ul>
   In addition, paths to the following tools must be provided as BreakSeq2 arguments:
   <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a>: Used for alignment to the breakpoint library</li>
      <li><a href="https://github.com/samtools/samtools">samtools</a>: Used for alignment filtering. Only version 0.1.19 has been tested.</li>
   </ul>
</p>


<h2>Installing BreakSeq2</h2>
<p>BreakSeq2 is a python package and can be installed using <code>pip</code>. To install type <code>pip install https://github.com/bioinform/breakseq2/archive/2.1-alpha.tar.gz</code>. The current version
of BreakSeq2 is 2.1-alpha. In general, the install source would be https://github.com/bioinform/breakseq2/archive/version.tar.gz</p>


<h2>Running BreakSeq2</h2>

<p>The following is the list of required options for BreakSeq2</p>
<ul>
    <li><code>--samtools path</code>: Path to samtools executable.</li>
    <li><code>--bwa path</code>: Path to BWA executable.</li>
    <li>Breakpoint library either as FASTA or GFF formats. The most up-to-date breakpoint library can be found <a href="http://sv.gersteinlab.org/phase1bkpts/breakseq2_bplib_20150129.zip" onlick="trackOutboundLink('http://sv.gersteinlab.org/phase1bkpts/breakseq2_bplib_20150129.zip'); return false;">here</a>. When the library is in GFF format, a corresponding insertion sequence file must also be present with the suffix .ins.</li>
    <li><code>--bams bam1 bam2 ...</code>: BAMs containing read alignments.</li>
    <li><code>--reference path</code>: Reference genome in FASTA format.</li>
</ul>

<p>Below are some examples of running BreakSeq2</p>

<h3>Running with breakpoint library as GFF</h3>

<p><code>run_breakseq2.py --reference b37.fasta --bams bwamem.bam --work work --bwa bwa-0.7.12/bwa --samtools samtools-0.1.19/samtools --bplib_gff bplib.gff --nthreads 4 --sample NA12878</code></p>
<p>Note that the sample name is not a required option. If it is missing, then the unique sample name is inferred from the BAMs. If there is no unique sample name and none is specified on the command-line, BreakSeq2 will fail. In the above example, since the breakpoint library was specified as a GFF, the insertion sequence file bplib.ins must also be present. BreakSeq2 will use the directory <code>work</code> for all the intermediate as well as the final output. Internally, BreakSeq2 uses the breakpoint
library in FASTA format, so the conversion from GFF to FASTA is done before the actual processing. BreakSeq2 uses parallelism whenever possible for best performance-in this case, 4 threads will be used for some steps of the processing.</p>

<h3>Running with breakpoint library as FASTA</h3>

<p><code>run_breakseq2.py --reference b37.fasta --bams bwamem.bam --work work --bwa bwa-0.7.12/bwa --samtools samtools-0.1.19/samtools --bplib bplib.fa --nthreads 4</code></p>
<p>Note that since the breakpoint library was specified as a FASTA file, bplib.ins is not required here since the insertion sequences are already accounted for in the FASTA.</p>

<p>In general, type <code>run_breakseq2.py -h</code> for a list of all command-line options.</p>


    </div>
    
    


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
</body>
